src/main/java/frc/robot/Robot.java

```diff
@@ -15,89 +47,157 @@
-  @Override
-  public void autonomousInit() {
-    m_autonomousCommand = m_robotContainer.getAutonomousCommand();
-
-    // schedule the autonomous command (example)
-    if (m_autonomousCommand != null) {
-      m_autonomousCommand.schedule();
+    private Command m_autonomousCommand;
+    private SendableChooser<Command> pathSelector;
+
+    private Intake intake;
+    private Swerve swerve;
+    private Shooter shooter;
+    private Climb climb;
+    private Wrist wrist;
+    private XboxOI oi;
+    private RobotState currState;
+    private AutoCommands auto;
+
+    @Override
+    public void robotInit() {
+        intake = Intake.getInstance();
+        climb = Climb.getInstance();
+        oi = new XboxOI();
+        swerve = Swerve.getInstance();
+        shooter = Shooter.getInstance();
+        wrist = Wrist.getInstance();
+        currState = RobotState.IDLE;
+
+        auto = new AutoCommands(swerve, 
+            transitionState(RobotState.INTAKE).andThen(new WaitUntilCommand(() -> intake.hasNote())).andThen(transitionState(RobotState.INDEX)), 
+            transitionState(RobotState.SUCK_IN),
+            transitionState(RobotState.FENDER).andThen(new WaitUntilCommand(() -> !shooter.hasNote())).andThen(transitionState(RobotState.IDLE))
+        );
+
+
+        configureBindings();
     }
-  }
-
-  /** This function is called periodically during autonomous. */
-  @Override
-  public void autonomousPeriodic() {}
-
-  @Override
-  public void teleopInit() {
-    // This makes sure that the autonomous stops running when
-    // teleop starts running. If you want the autonomous to
-    // continue until interrupted by another command, remove
-    // this line or comment it out.
-    if (m_autonomousCommand != null) {
-      m_autonomousCommand.cancel();
+
+
+    private void configureBindings() {
+        // Driver
+        oi.getDriverTrigger(Controls.INTAKE).onTrue(transitionState(RobotState.INTAKE));
+        oi.getDriverButton(Controls.REVERSE).onTrue(transitionState(RobotState.SUCK_IN).andThen(transitionState(RobotState.IDLE)));
+        oi.getDriverButton(Controls.AMP).onTrue(transitionState(RobotState.AMP));
+        oi.getDriverTrigger(Controls.FENDER).onTrue(transitionState(RobotState.FENDER));
+        oi.getDriverButton(Controls.REVERSE).onTrue(transitionState(RobotState.REVERSE)).onFalse(transitionState(RobotState.IDLE));
+
+        oi.getDriverButton(Controls.IDLE).onTrue(transitionState(RobotState.IDLE));
+
+
+        // Operator
+        oi.getOperatorButton(Controls.CLIMB_UP).onTrue(climb.moveUp()).onFalse(climb.stop());
+        oi.getOperatorButton(Controls.CLIMB_DOWN).onTrue(climb.moveDown()).onFalse(climb.stop());
+        oi.getOperatorButton(Controls.OPERATOR_REVERSE).onTrue(transitionState(RobotState.SUCK_IN));
+
+        oi.getOperatorButton(Controls.ZERO_GYRO).onTrue(swerve.zeroGyro());
+        oi.getOperatorButton(Controls.ZERO_ENCODERS).onTrue(new InstantCommand(() -> swerve.updateEncoders()));
```

Comment:
this can probably be moved in PopLib, create a new method in swerve class that uses functional commands